src/main/java/de/teamlapen/vampirism/entity/converted/BiteableEntryManager.java

```diff
@@ -94,6 +97,57 @@ BiteableEntry calculate(@Nonnull CreatureEntity creature) {
         }
     }
 
+    /**
+     * checks if the entity type is blacklisted for blood calculation
+     * does not check if the entity is blacklisted in the server config ({@link #isEntityTypeBlacklisted(EntityType)})
+     *
+     * @param type the entity type
+     * @returns weather the entity type is blacklisted or not
+     */
+    public boolean isEntityTypeBlacklistedByDefault(EntityType<?> type) {
+        if (!VampirismConfig.SERVER.autoCalculateEntityBlood.get()) return true;
+        if (!(type.getClassification() != EntityClassification.MONSTER && type.getClassification() != EntityClassification.WATER_CREATURE))
+            return true;
+        if (ModTags.Entities.VAMPIRE.contains(type)) return true;
+        return false;
+    }
+
+    /**
+     * checks if the entity type is blacklisted for blood calculation
+     *
+     * @param type the entity type
+     * @returns weather the entity type is blacklisted or not
+     */
+    public boolean isEntityTypeBlacklisted(EntityType<?> type) {
+        if (isEntityTypeBlacklistedByDefault(type)) return true;
+        if (isConfigBlackListed(type.getRegistryName())) return true;
+        return false;
+    }
+
+    /**
+     * checks if the creature entity is blacklisted
+     *
+     * @param creature the entity to check
+     * @returns weather the entity is blacklisted or not
+     */
+    private boolean isEntityBlacklisted(CreatureEntity creature) {
+        if (isEntityTypeBlacklisted(creature.getType())) return true;
+        if (!(creature instanceof AnimalEntity)) return true;
```

Comment:
Move instanceof checks to the top since they are likely much less costly than the method and probably have the only a slighlty smaller chance of returning true

Reply:
moved